console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,26 @@ type ConsolePluginSpec struct {
 	// i18n is the configuration of plugin's localization resources.
 	// +optional
 	I18n ConsolePluginI18n `json:"i18n"`
+	// csp is a list of Content Security Policy directives for the plugin.
+	CSP []ConsolePluginCSP `json:"csp"`
+}
+
+// DirectiveType is an enumeration of supported CSP directives
+type DirectiveType string
+
+const (
+	DefaultSrc DirectiveType = "default-src"
+	ScriptSrc  DirectiveType = "script-src"
+	ImgSrc     DirectiveType = "img-src"
+	StyleSrc   DirectiveType = "style-src"
+	FontSrc    DirectiveType = "font-src"
+	BaseUri    DirectiveType = "base-uri"
+)
+
+// ConsolePluginCSP
+type ConsolePluginCSP struct {
+	Directive DirectiveType `json:"directive"`
```

Comment:
I like the idea of making this generic to support other directives as long as we have validation that only the ones we want can be customized. (Plugins shouldn't be able to set any directive.)

Reply:
@spadgett not sure what you mean here. Could you please be more specific?

Reply:
Just a general comment that I agree with the approach :) But we should be careful what directives we allow plugins to change.

Reply:
@spadgett this enumeration makes sure that only these types are valid.

Reply:
Right. My point is that we should be deliberate in what we add to this enum in the future.